lib/view_model/events.dart

```diff
@@ -58,22 +58,23 @@ class EventsLogic extends ChangeNotifier {
     for (int i = 0; i < eventsDocIds.length; i++) {
       DocumentSnapshot<Object?> snap =
           await _collectionRef.doc(eventsDocIds[i]).get();
-      addInSpecificEvents(snap);
+      _specificEvents.add(addInSpecificEvents(snap));
     }
     stateOfSpecificEvents = StateOfSpecificEvents.loaded;
     notifyListeners();
   }
 
-  addInSpecificEvents(DocumentSnapshot<Object?> snap) {
-    _specificEvents.add(Events(
+ Events addInSpecificEvents(DocumentSnapshot<Object?> snap) {
+    return Events(
```

Comment:
Add a fromJson method instead of using this constructor for DB data
See [here](https://flutter.dev/docs/development/data-and-backend/json)